### ISSUE TITLE
Make hotkey for switching to "View > Show all branches" configurable

### ISSUE THREAD

**Issue by per1234** [gitextensions/gitextensions#8453]
## Feature description

Please add the hotkey ("Esc") that switches from the **View > Show current branch only** or **View > Show filtered branches** selection to **View > Show all branches** option to the configurable hotkeys in **Tools > Settings > Hotkeys**.

I find it very confusing to have my repository view change unexpectedly. This has caused me problems several times when I didn't notice the view had changed and assumed the commit at the top of the list was the tip of the currently selected branch.

Because this hotkey is not shown in the meny, I only discovered today, after a long time struggling with unexpected changes to the setting, that it was the Esc key causing this.

## Environment

Tested with the latest CI build to verify the behavior is current:

- Git Extensions 4.0.0.10126
- Build ffd546cac90a008e615ad203614b786e55ca034f (Dirty)
- Git 2.27.0.windows.1
- Microsoft Windows NT 10.0.18362.0
- .NET Framework 4.8.4220.0
- DPI 96dpi (no scaling)

As well as the latest release version I use normally:

- Git Extensions 3.4.3.9999
- Build d4b0f48bbf3e39a71f5d7ff5231be5678d4aa0f1
- Git 2.27.0.windows.1
- Microsoft Windows NT 10.0.18362.0
- .NET Framework 4.8.4220.0
- DPI 96dpi (no scaling)

**Comment by RussKie** [comment]
> Please add the hotkey ("Esc") that switches from the **View > Show current branch only** or **View > Show filtered branches** selection to **View > Show all branches** option to the configurable hotkeys in **Tools > Settings > Hotkeys**.

Wow, 7+ years using the app and I never knew we had this behaviour 😮

**Comment by mstv** [comment]
First, `Esc` clears the `Filter:` text, then the `Branches:` selection.
Would a tooltip on these two boxes suffice?

BTW `Filter:` takes precedence over `Branches:`.

![grafik](https://user-images.githubusercontent.com/36601201/94065825-ea674600-fdeb-11ea-9849-af79a9e783c3.png)

**Comment by per1234** [comment]
No, that wouldn't suffice for me. 

It wouldn't have helped with my confusion because I would never have thought to hover the mouse pointer over those boxes. Theres no obvious connection between the **View > Show all branches** option and those filters. I never use those, so they are always blank, thus there's not even a visible change to them.

I wouldn't help with my ongoing problem of unintentionally triggering this with the Esc key.

---
What would help (but still not suffice for me), is if there was a way to disable that "quick search" thing (https://github.com/gitextensions/gitextensions/issues/8474). The most common reason (but not the only reason) for me to press the "Esc" key while using Git Extensions is trying to get out of the quick search box (I was forced to set it to a huge timeout value to mitigate the harmful effect of quick search unexpectedly jumping me to a different commit).

It's especially important to provide control over these single keypress hotkeys (quick search is essentially a universal single keypress hotkey). Combination hotkeys are far less likely to be triggered unintentionally.

**Comment by mstv** [comment]
Indeed, at least the second hit of <kbd>Esc</kbd> has the same effect as <kbd>Shift</kbd>+<kbd>Ctrl</kbd>+<kbd>A</kbd>.
The information could be added to the menu items (in the translation), e.g.:

![grafik](https://user-images.githubusercontent.com/36601201/94191300-9886f480-fead-11ea-8a9a-a0d1689ca60e.png)

In order to allow disabling the hotkey, we could add it as an official hotkey. Though this would complicate the above menu text extension. And:

In order to reduce the confusion caused by attempting to cancel Quick Search using <kbd>Esc</kbd>, we should implement it for Quick Search really, too. I guess then we will not need an explicit hotkey.

**Comment by RussKie** [comment]
May be can remove the filter branches toolstrip box completely and offer a better UX via the left panel?
That is, allow "show this branch only" as a context menu (this we already have) and allow branch multi-selection with a context menu - "show only these branches". Both options for local branches only.

I'd also remove <kbd>ESC</kbd> shortcut, because IMO it is unexpected in this context. Generally <kbd>ESC</kbd> is used to cancel an ongoing operation, and filter is change of state. E.g. looks at Excel, for example, once you have applied a filter it doesn't get reset via <kbd>ESC</kbd>.
